src/components/code-diff-editor/code-diff-editor.tsx

```diff
@@ -0,0 +1,223 @@
+   * Specifies if the editor should be readonly.
+   */
+  @Prop({ attribute: "readonly" }) readonly readonly: boolean = true;
+  @Watch("readonly")
+  readonlyChanged(newReadonly: boolean) {
+    if (this.#monacoDiffEditorInstance) {
+      this.#monacoDiffEditorInstance.updateOptions({
+        originalEditable: !newReadonly ?? true,
+        readOnly: newReadonly ?? false
+      });
+    }
+  }
+
+  /**
+   * Specifies the original value of the diff editor.
+   */
+  @Prop() readonly value: string;
+  @Watch("value")
+  valueChange(newValue: string) {
+    if (this.#monacoDiffEditorInstance) {
+      this.#monacoDiffEditorInstance.getModel()!.original.setValue(newValue);
+    }
+  }
+
+  /**
+   * Specifies the modified value of the diff editor.
+   */
+  @Prop({ attribute: "modified-value" }) readonly modifiedValue: string;
+  @Watch("modifiedValue")
+  modifiedValueChange(newModifiedValue: string) {
+    if (this.#monacoDiffEditorInstance) {
+      this.#monacoDiffEditorInstance
+        .getModel()!
+        .modified.setValue(newModifiedValue);
+    }
+  }
+
+  /**
+   * Specifies the schema URI for the YAML language.
+   */
+  @Prop() readonly yamlSchemaUri: string = "";
+  @Watch("yamlSchemaUri")
+  yamlSchemaUriChange(newUri: string) {
+    if (this.language !== "yaml") {
+      return;
+    }
+
+    // Necessary to not combine the current scheme with the new scheme
+    this.#editorId = `ch-diff-editor-${autoId++}`;
+
+    if (newUri) {
+      configureMonacoYaml(monaco, {
+        enableSchemaRequest: true,
+        format: true,
+        schemas: this.#getYamlSchemas()
+      });
+    } else {
+      configureMonacoYaml(monaco, {
+        enableSchemaRequest: true,
+        format: true,
+        schemas: []
+      });
+    }
+
+    this.#monacoDiffEditorInstance.setModel({
+      original: monaco.editor.createModel(
+        this.value,
+        this.language,
+        monaco.Uri.parse(`file:///${this.#editorId}.txt`)
+      ),
+      modified: monaco.editor.createModel(
+        this.modifiedValue,
+        this.language,
+        monaco.Uri.parse(`file:///${this.#editorId}-modified.txt`)
+      )
+    });
+  }
+
+  #getYamlSchemas = () => [
+    {
+      // If YAML file is opened matching this glob
+      fileMatch: [`**/${this.#editorId}.*`],
+      // Then this schema will be downloaded from the internet and used.
+      uri: this.yamlSchemaUri
+    },
+    {
+      // If YAML file is opened matching this glob
+      fileMatch: [`**/${this.#editorId}-modified.*`],
+      // Then this schema will be downloaded from the internet and used.
+      uri: this.yamlSchemaUri
+    }
+  ];
+
+  #setupDiffEditor() {
+    this.#monacoDiffEditorInstance = monaco.editor.createDiffEditor(
+      this.#monacoRef,
+      {
+        ...this.options,
+        ...(!(
+          "originalEditable" in this.options || "readOnly" in this.options
+        ) && {
+          originalEditable: !this.readonly,
+          readOnly: this.readonly
+        })
```

Comment:
I think is better to use the readonly property as the default value, even if one of the properties (`originalEditable` or `readOnly`) is defined. Otherwise, we have to specify the current behavior in the description of the `readonly` property.

```suggestion
        originalEditable: this.options.originalEditable ?? !this.readonly,
        readOnly: this.options.readOnly ?? this.readonly
```